### ISSUE TITLE
Attention not fused

### ISSUE THREAD

**Issue by jxchenus** [microsoft/onnxruntime#10359]
**Describe the bug**
onnxruntime.transformers.optimizer reports "Attention not fused" for tf2onnx converted bert-large-uncased-whole-word-masking-finetuned-squad model included in Tensorflow_Tf2onnx_Bert-Squad_OnnxRuntime_CPU.ipynb.

**Urgency**
As soon as possible.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux and Mac
- ONNX Runtime installed from (source or binary): both source or binary
- ONNX Runtime version: 1.9.0 and ort-nightly 1.11.0.dev20220120004
- Python version: 3.7 and 3.8
- Visual Studio version (if applicable): N/A
- GCC/Compiler version (if compiling from source): gcc (GCC) 7.3.1 
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**To Reproduce**
- Use Jupyter notebook to run onnxruntime/python/tools/transformers/notebooks/Tensorflow_Tf2onnx_Bert-Squad_OnnxRuntime_CPU.ipynb
- After completing cell [14], get two models under onnxruntime/python/tools/transformers/notebooks/onnx_models
- The optimized model (tf2onnx_bert-large-uncased-whole-word-masking-finetuned-squad_opt_cpu.onnx) does not contain "Attention" node and the inference time in cell [15] is about the same as in cell [10]
- Explicitly invoke optimizer.py on the original model "tf2onnx_bert-large-uncased-whole-word-masking-finetuned-squad.onnx" and got:

remove_useless_cast_nodes: Removed 0 Cast nodes with output type same as input
     remove_identity: Removed Identity count: 0
   process_embedding: start processing embedding layer...
        skip_reshape: Skip consequent Reshape count: 24
calling fuse attention
      fuse_attention: Fused Attention count:0
         prune_graph: Graph pruned: 0 inputs, 0 outputs and 0 nodes are removed
            optimize: opset verion: 13
  save_model_to_file: Sort graphs in topological order
  save_model_to_file: Model saved to /home/chejane/projects/asrct-386/bert/tf2onnx_bert-cpu_opt.onnx
get_fused_operator_statistics: Optimized operators:{'EmbedLayerNormalization': 0, 'Attention': 0, 'Gelu': 0, 'FastGelu': 0, 'BiasGelu': 0, 'LayerNormalization': 0, 'SkipLayerNormalization': 0}
  is_fully_optimized: Attention not fused
                main: The model has been optimized.

**Expected behavior**
Expect the optimizer to fuse the attention component nodes into attention nodes.

**Screenshots**
N/A

**Additional context**
The cause of the issue is the optimizer is looking for SkipLayerNormalization and LayerNormalization nodes in the tf_bert model:

https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/tools/transformers/onnx_model_bert_tf.py#L284

but these two types of nodes get converted to something else by tf2onnx converter, so they don't exist in the converted model

**Comment by tianleiwu** [comment]
@wangyems to take a look.

**Comment by wangyems** [comment]
Hi @jxchenus, this tutorial was drafted with the priority of introducing tf2onnx. It's expected that the optimizer cannot fuse Layernorm, Gelu, Attention... etc. It's in our backlog and currently there is no ETA. 
Meanwhile, you can refer to [developer guide](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/tools/transformers/Dev_Guide.md) for a better idea on how offline optimizer works and make any necessary changes to fuse the graph on your own.
Another way is to [convert the tensorflow model to pytorch](https://huggingface.co/transformers/v1.0.0/converting_tensorflow_models.html) and then [export the pytorch model to onnx](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/tools/transformers/notebooks/PyTorch_Bert-Squad_OnnxRuntime_CPU.ipynb)

**Comment by wangyems** [comment]
> Thanks @wangyems for the quick response.
> 
> It would be nice to make it more obvious in the notebook Tensorflow_Tf2onnx_Bert-Squad_OnnxRuntime_CPU.ipynb that the optimizer is not expected to fuse the attention layer and produce any performance gain.

You're right. This is sort of misleading at the moment.
